user/src/com/google/gwt/user/client/ui/RootPanel.java

```diff
@@ -179,6 +179,12 @@ public static RootPanel get(String id) {
       // for any reason, return a new RootPanel rather than the existing one (
       // see issue 1937).
       if ((elem == null) || (rp.getElement() == elem)) {
+        // if id and elem are null, set the element of existing rp with the 
+        // current body to handle scenarios where body gets replaced/refreshed
+        // from DOM on certain actions like page navigation.
+        if(id == null && elem == null) {
+          rp.setElement((Element) natGetBodyElement());
+        }
```

Comment:
This should probably rather avoid entering this `if` block when `rp.getElement() != getBodyElement()` and thus eventually return a new `DefaultRootPanel`. Using `setElement` on an existing widget that already has an element (like a `DefaultRootPanel`) is plain wrong (and will throw an assertion error in dev mode).

(not to mention that this code doesn't even compile as `natGetBodyElement()` doesn't exist)

Reply:
My bad ! I have replaced with getBodyElement() . 
The if check at line 181 first check for (elem == null) which is true and does not validate if element matches with the body or not . So added this check to keep the rootPanel same with the correct body reference

Reply:
If `elem` is non-null, then the `if` will continue with the `rp.getElement() == elem` check, and then if `id` happens to be null, the rootpanel will get its element replaced... with its current element. Even though that is the same element, in dev mode `setElement()` will still fail.